### PR TITLE
Ignore Intellij IDEA files and some others

### PR DESCRIPTION
**Pull request by breeswish**
The new `.gitignore` is taken from TiKV, which additionally ignores Intellij IDEA and some other common files.

### CODE REVIEW

**Comment by siddontang** [comment]
seem it is not common to track the Cargo.lock in a library, we should remove it.

**Comment by breeswish** [comment]
@siddontang Oh, got it. Removed.

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,28 @@
...
+.idea/
+*.iml
+out/
+
+# Vscode files
+.vscode/**
+
 target
+tmp
+/bin
```
Comment (siddontang) [inline .gitignore]: why remove `bin` but add `/bin` here?

Comment (breeswish) [reply .gitignore]: Copied from TiKV. We only need to ignore the `bin` directory in the project root.

**Pull request merged**